src/draw/lv_image_decoder.h

```diff
@@ -111,10 +112,20 @@ typedef struct _lv_image_decoder_t {
     lv_image_decoder_open_f_t open_cb;
     lv_image_decoder_get_area_cb_t get_area_cb;
     lv_image_decoder_close_f_t close_cb;
-    uint32_t cache_data_type;
+
+    lv_cache_free_cb_t cache_free_cb;
     void * user_data;
 } lv_image_decoder_t;
 
+typedef struct _lv_image_decoder_cache_data_t {
+    const void * src;
+    lv_image_src_t src_type;
+
+    const lv_draw_buf_t * decoded;
+    const lv_image_decoder_t * decoder;
```

Comment:
Could it be destroyed after data added to cache?

Reply:
It's a global instance